src/Parser.js

```diff
@@ -43,8 +43,8 @@ export function parse (value, root, parent, rule, rules, rulesets, pseudo, point
 		switch (previous = character, character = next()) {
 			// (
 			case 40:
-				if (previous != 108 && charat(characters, length - 1) == 58) {
-					if (indexof(characters += replace(delimit(character), '&', '&\f'), '&\f') != -1)
+				if (charat(characters, length - 1) == 58) {
+					if (indexof(characters += replace(delimit(character), '&', '&\f'), '&\f', abs(points[index - 1])) != -1)
```

Comment:
iiuc this should limit the search to the portion of the text that contains the current selector. One potential problem I see here is that it might result in out-of-bounds access on `points`. Is this something to worry about at all from the engines' PoV?

Reply:
yes, out of bound access would be slower in the js engine

```js
// (
			case 40:
				if (previous != 108 && charat(characters, length - 1) == 58) {
					if (indexof(characters += replace(delimit(character), '&', '&\f'), '&\f', abs(index ? points[index - 1] : 0)) != -1)
						ampersand = -1
					break
				}
```

Reply:
> also garbage data in points array if it's not in a comma separated selector so we should check if it is before.

I don't quite understand this. My understanding of this parser is quite superficial ;p especially with such long breaks between my small tweaks here and there. I'm not sure in what situation some garbage could end up in `points` here.

Reply:
`points` is never cleared/zero'ed out between function calls, it act's like a reusable stack(we avoid growing it if need be), as such reading any random index(in this case zero) into it could give you some random value from a previous stack call if you aren't sure that `points` is being used in the current stack call, the `points[index++] = ...`

Reply:
ok, it makes more sense now